src/snowflake/snowpark/session.py

```diff
@@ -441,18 +440,48 @@ def sql_simplifier_enabled(self) -> bool:
         return self._sql_simplifier_enabled
 
     @property
-    def custom_packages_upload_enabled(self) -> bool:
-        """Set to ``True`` to automatically upload custom packages.
-        If set to ``True``, any package mentioned during UDF or stored procedure registration, or via :func:`add_packages`  or :func:`add_requirements`, will be installed by pip, zipped and made available as an import via a remote stage.
-        Note that this ``Session`` parameter is **experimental**, please do not use it in production!"""
-        return self._custom_packages_upload_enabled
+    def custom_package_usage_config(self) -> Dict:
```

Comment:
Documentation
![doc_1](https://github.com/snowflakedb/snowpark-python/assets/135074967/22165400-d07f-4570-a9c7-b1aa81c6cab4)
![doc_2](https://github.com/snowflakedb/snowpark-python/assets/135074967/c9171df6-462a-41b1-b741-fea9c36c6c8d)

Reply:
Hi @sfc-gh-straut , Could I have a review of this

Reply:
Looks great, Vivek. One nit -- in your last section, the note, I would use a bulleted list rather than a numbered list. The two items in the list don't seem to represent a sequence for which you'd need numbers. They're just two items.